src/dialogs/sidebar/dialog-edit-sidebar.ts

```diff
@@ -22,21 +29,43 @@ class DialogEditSidebar extends LitElement {
 
   @query("ha-md-dialog") private _dialog?: HaMdDialog;
 
-  @state() private _order: string[] = [];
+  @state() private _order?: string[];
 
-  @state() private _hidden: string[] = [];
+  @state() private _hidden?: string[];
 
-  private _saveCallback?: (order: string[], hidden: string[]) => void;
+  @state() private _error?: string;
 
-  public async showDialog(params: EditSidebarDialogParams): Promise<void> {
+  /**
+   * If user has old localStorage values, show a confirmation dialog
+   */
+  private _migrateToUserData = false;
+
+  public async showDialog(): Promise<void> {
     this._open = true;
 
-    const storedOrder = localStorage.getItem("sidebarPanelOrder");
-    const storedHidden = localStorage.getItem("sidebarHiddenPanels");
+    this._getData();
+  }
 
-    this._order = storedOrder ? JSON.parse(storedOrder) : this._order;
-    this._hidden = storedHidden ? JSON.parse(storedHidden) : this._hidden;
-    this._saveCallback = params.saveCallback;
+  private async _getData() {
+    try {
+      const data = await fetchFrontendUserData(this.hass.connection, "sidebar");
+      this._order = data?.panelOrder;
+      this._hidden = data?.hiddenPanels;
+
+      // fallback to old localStorage values
+      if (!this._order) {
+        const storedOrder = localStorage.getItem("sidebarPanelOrder");
+        this._migrateToUserData = !!storedOrder;
+        this._order = storedOrder ? JSON.parse(storedOrder) : [];
+      }
+      if (!this._hidden) {
+        const storedHidden = localStorage.getItem("sidebarHiddenPanels");
+        this._migrateToUserData = !!storedHidden;
```

Comment:
```suggestion
        this._migrateToUserData = this._migrateToUserData || !!storedHidden;
```
Not sure it is a real case but this change would handle when 'sidebarPanelOrder' is not empty and 'sidebarHiddenPanels' is empty